public/app/features/explore/ExploreActions.tsx

```diff
@@ -0,0 +1,64 @@
+import { FC, useEffect, useState } from 'react';
+import { useRegisterActions, useKBar, Action } from 'kbar';
+import { useDispatch, useSelector } from 'react-redux';
+import { StoreState, ExploreId } from 'app/types';
+import { splitOpen, splitClose } from './state/main';
+import { isSplit } from './state/selectors';
+import { runQueries } from './state/query';
+
+interface Props {
+  exploreId: ExploreId;
+}
+
+export const ExploreActions: FC<Props> = ({ exploreId }: Props) => {
+  const [actions, setActions] = useState<Action[]>([]);
+  const { query } = useKBar();
+  const dispatch = useDispatch();
+  const { splitted } = useSelector((state: StoreState) => {
+    const splitted = isSplit(state);
+    return {
+      splitted,
+    };
+  });
+
+  useEffect(() => {
+    const actionsArr: Action[] = [
+      {
+        id: 'explore/run-query',
```

Comment:
given this component is rendered for each explore pane and this action has a static id, the second time it registers it overrides the previous one. In Split mode this mean we'll be able to run only the query on the right pane.